scripts/keras_classification.py

```diff
@@ -0,0 +1,738 @@
+                        actual initial value used should be set by the
+                        sparseml recipe
+  --optim-args OPTIM_ARGS
+                        Additional args to be passed to the optimizer passed
+                        in as a json object
+#############
+EXAMPLE: Fine-tune a ResNet50 model from Keras Application, pretrained on Imagenet,
+for the Imagenette dataset. The recipe could be used to define learning rate schedule.
+Note that there're no pruning performed from this experiment.
+python scripts/keras_classification.py train \
+     --arch-key keras_applications.ResNet50 \
+     --pretrained-dataset imagenet \
+     --dataset imagenette \
+     --dataset-path <DATASET_DIR> \
+     --train-batch-size 64 \
+     --recipe-path <RECIPE_PATH>
+############
+EXAMPLE: Assuming that a ResNet50 model from Keras Application was fine-tuned for
+the Imagenette dataset. The following command specifies the model checkpoint after
+fine-tuning, and a recipe that could be used next to prune its weights:
+python scripts/keras_classification.py train \
+     --arch-key keras_applications.ResNet50 \
+     --checkpoint-path <MODEL_CHECKPOINT> \
+     --dataset imagenette \
+     --dataset-path /hdd/datasets/imagenette \
+     --train-batch-size 64 \
+     --recipe-path <RECIPE_PATH>
+
+In the above command, specifying "zoo" for --checkpoint-path forces loading the
+base model from SparseZoo. The following command prunes a ResNet50 from Keras
+Applications fine-tuned for Imagenette dataset, using a recipe from SparseZoo:
+
+python scripts/keras_classification.py train \
+     --arch-key keras_applications.ResNet50 \
+     --checkpoint-path zoo \
+     --pretrained-dataset imagenette \
+     --dataset imagenette \
+     --dataset-path <DATASET_DIR> \
+     --train-batch-size 64 \
+     --recipe-path zoo:cv/classification/resnet_v1-50/keras/keras.applications/imagenette/pruned-moderate?recipe_type=original
+############
+EXAMPLE: The following command loads a ResNet50 model from a checkpoint, then
+evaluate it on the validation of the Imagenette dataset.
+python scripts/keras_classification.py evaluate \
+     --arch-key keras_applications.ResNet50 \
+     --checkpoint-path <MODEL_CHECKPOINT> \
+     --dataset imagenette \
+     --dataset-path /hdd/datasets/imagenette \
+     --test-batch-size 64
+############
+EXAMPLE: The following command loads a ResNet50 model from a checkpoint, then export it
+into ONNX format. It also generates a number of inputs, outputs and labels.
+python scripts/keras_classification.py export \
+     --arch-key keras_applications.ResNet50 \
+     --checkpoint-path <MODEL_CHECKPOINT> \
+     --dataset imagenette \
+     --dataset-path /hdd/datasets/imagenette \
+     --num-samples <NUM_SAMPLES>
+     --save-dir <EXPORT_DIR>
+"""
+
+import argparse
+import json
+import math
+import os
+from typing import Optional, Tuple
+
+import numpy as np
+import tensorflow as tf
+from tensorflow import keras
```

Comment:
to make this properly compatible we should use our keras import that checks between tensorflow.keras and native keras